concent_api/core/validation.py

```diff
@@ -218,3 +226,121 @@ def validate_list_task_to_compute_ids(subtask_results_accepted_list):
+
+    if file_transfer_token.storage_cluster_address != settings.STORAGE_CLUSTER_ADDRESS:
+        return ('This token does not allow file transfers to/from the cluster you are trying to access.', ErrorCode.MESSAGE_STORAGE_CLUSTER_WRONG_CLUSTER)
+
+    # -CLIENT_PUBLIC_KEY
+    if not isinstance(file_transfer_token.authorized_client_public_key, bytes):
+        return ('Wrong type of authorized_client_public_key field value.', ErrorCode.MESSAGE_AUTHORIZED_CLIENT_PUBLIC_KEY_WRONG_TYPE)
+
+    # -FILES
+    if not all(isinstance(file, dict) for file in file_transfer_token.files):
+        return ('Wrong type of files field value.', ErrorCode.MESSAGE_FILES_WRONG_TYPE)
+
+    transfer_token_paths_to_files = [file["path"] for file in file_transfer_token.files]
+    if len(transfer_token_paths_to_files) != len(set(transfer_token_paths_to_files)):
+        return ('File paths in the token must be unique', ErrorCode.MESSAGE_FILES_PATHS_NOT_UNIQUE)
+
+    file_checksums = [file["checksum"] for file in file_transfer_token.files]
+    for file_checksum in file_checksums:
+        if not isinstance(file_checksum, str):
+            return ("'checksum' must be a string.", ErrorCode.MESSAGE_FILES_CHECKSUM_WRONG_TYPE)
+
+        if len(file_checksum) == 0 or file_checksum.isspace():
+            return ("'checksum' cannot be blank or contain only whitespace.", ErrorCode.MESSAGE_FILES_CHECKSUM_EMPTY)
+
+        if ":" not in file_checksum:
+            return ("'checksum' must consist of two parts separated with a semicolon.", ErrorCode.MESSAGE_FILES_CHECKSUM_WRONG_FORMAT)
+
+        if not file_checksum.split(":")[0] == set(HashingAlgorithm._value2member_map_).pop():
+            return ("One of the checksums is from an unsupported hashing algorithm.", ErrorCode.MESSAGE_FILES_CHECKSUM_INVALID_ALGORITHM)
+
+        assert set(HashingAlgorithm) == {HashingAlgorithm.SHA1}, "If you add a new hashing algorithms, you need to add validations below."
+        if VALID_SHA1_HASH_REGEX.fullmatch(file_checksum.split(":")[1]) is None:
+            return ("Invalid SHA1 hash.", ErrorCode.MESSAGE_FILES_CHECKSUM_INVALID_SHA1_HASH)
+
+    file_sizes = [file["size"] for file in file_transfer_token.files]
+    for file_size in file_sizes:
+        if file_size is None:
+            return ("'size' must be an integer.", ErrorCode.MESSAGE_FILES_SIZE_EMPTY)
+
+        try:
+            int(file_size)
+        except (ValueError, TypeError):
+            return ("'size' must be an integer.", ErrorCode.MESSAGE_FILES_SIZE_WRONG_TYPE)
+
+        if int(file_size) < 0:
+            return ("'size' must not be negative.", ErrorCode.MESSAGE_FILES_SIZE_NEGATIVE)
+
+    # Validate category in FileInfo
+    assert all('category' in file for file in file_transfer_token.files)
+    assert all(isinstance(file['category'], FileTransferToken.FileInfo.Category) for file in file_transfer_token.files)
+
+    categories = [file_info['category'] for file_info in file_transfer_token.files]
```

Comment:
I think I have already seen it somwhere... Could you please make sure it is not a duplication (and extract/reuse if necessary)?

Reply:
forget it ;)